PR to start TSO notebook review

nespinoza opened the pull request:
Hi folks!

 Here's a PR to perform some TSO notebook reviews. So far, I have:

- **[Feb 9, 2022] Changed the input folders.** Prior and on the JWebbinar, they pointed to the environment defaults. Now the users' working directory.

ojustino commented:
Hi @nespinoza,

Thank you for submitting these changes to your notebook. Please read on for your technical review.

## Before you begin

The technical review helps ensure that contributed notebooks a) run from top to bottom, b) follow the [PEP8 standards](https://www.python.org/dev/peps/pep-0008/) for Python code readability, and c) conform to the Institute's [style guide](https://github.com/spacetelescope/style-guides/blob/master/guides/jupyter-notebooks.md) for Jupyter Notebooks.

I've pushed the review as a new commit in this pull request. **To view and edit the commit locally, follow these steps:**

```
git checkout tso-changes
git fetch YOUR-REMOTE-FORK
git merge YOUR-REMOTE-FORK/tso-changes
```

_(`YOUR-REMOTE-FORK` is your fork's online copy. It's often `origin`, but if  you don't know your name for it, run `git remote -v` and choose the one whose path ends with `YOUR-USERNAME/jwebbinar_prep.git`.)_

From here you can work on your branch as normal. If you have trouble with this step, please let me know before continuing.

---

## Instructions

After updating your local copy of this branch, **please open your notebook and address any warnings or errors you find**.

If you see cells with output like this, it means some of your code doesn't follow the PEP8 standards of code readability:

<img width="574" alt="image" src="https://user-images.githubusercontent.com/12895749/121729210-306c5300-cabc-11eb-90eb-eb494dca53c4.png">

_(In the example above, `INFO - 3:3: E111` means that the text entered on line 3 at index 3 caused the warning "E111". The violation is briefly described at the end of the message.)_

You can test that your edits satisfy the standard by installing `flake8` on the command line with:
```
pip install flake8==3.7.3 pycodestyle_magic
```

Then, restart the notebook and run the following cells:

<img width="580" alt="image" src="https://user-images.githubusercontent.com/12895749/121743209-fd7f8a80-cace-11eb-86a5-90e7b857a8be.png">

After that, edit and re-run cells with warnings until you've fixed all of them. **Please remember to delete the cells shown in the above image before pushing your changes back to this pull request.**

**If you have questions or feedback on specific cells, click the earlier message in this thread from the "review-notebook-app" bot.** There, you can comment on specific cells and view what's changed in the new commit. I may also write comments there. Anything posted there will also be reflected in this pull request's conversational thread.

## The three-point review (\*action required\*)

1. ✅/⚠️ Notebook execution: **I was able to run the notebook from top to bottom.**
    - I will leave a comment on a cell with a plot that could use a different aspect ratio.

2. ⚠️ Code style
    - Due to the length of the notebook, there are several PEP8 warnings to address. The advice in the "__Instructions__" section should help in getting started.
    - The cells I added at the top of the notebook include code for muting other packages' warnings, but `jwst.pipeline` objects don't respect it. This may make it harder for you to distinguish PEP8 warnings from JWST pipeline logging messages as you edit the cells. My best advice is that the PEP8 warnings typically come at the end of a cell's output. As such, any cell whose last outputs don't match the format from the first image in the above "__Instructions__" section should be good to go.

3.  ✅/⚠️ Notebook style
    - The notebook is well-documented and mostly adheres to the style guide. There is some information in the header that normally goes in the footer, but we can discuss that in more detail once the PEP8 warnings have been addressed.

ojustino commented on tso_session/ExploringTSOProducts.ipynb:
```diff
@@ -114,15 +114,87 @@
     "\n",
```
<p>You can ignore the first two warnings here; <code>flake8</code>  isn't recognizing that the <code>%set_env</code>  statements are not Python.</p><p> </p><p>You can get rid of these warnings by either separating environment variable assignment into its own cell or by moving it after the imports in this one.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/spacetelescope/jwebbinar_prep/pull/107/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

ojustino replied:
<pre class="ql-syntax" spellcheck="false">Line #1.    fig, ax1 = plt.subplots(figsize=(30, 5))
</pre><p>I found a figsize between (15, 5) and (20, 5) to be more readable across a variety of screen resolutions, but will leave the final choice up to you.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/spacetelescope/jwebbinar_prep/pull/107/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='173'/>

nespinoza commented:
Hi @ojustino! Here are my changes:

1. Added most of your suggestions _except_ the PEP8 fixes (see below).
2. Modified `datamodels.open` with `datamodels.RampModel` which is prefered for actual JWST data.

I spoke with Cami (@camipacifici), and mentioned to her that I'm quite busy to work on the PEP8 stuff. We both decided it was pretty minor and that we should be OK. Can you confirm Cami?

Thanks for your review!
Néstor

camipacifici commented:
I approve this one. I will open an issue so we do not forget the PEP8 fixes.
